publication-rest/src/main/java/no/unit/nva/publication/fetch/FetchPublicationHandler.java

```diff
@@ -117,10 +117,14 @@ protected String processInput(Void input, RequestInfo requestInfo, Context conte
     }
 
     private String producePublicationResponseWhenUnpublished(RequestInfo requestInfo, Resource resource)
-        throws GoneException {
-        return userCanUpdateResource(requestInfo, resource)
-                   ? createPublicationResponse(requestInfo, resource)
-                   : produceRemovedPublicationResponse(resource, requestInfo);
+        throws GoneException, UnsupportedAcceptHeaderException {
+        return shouldRedirectToDuplicate(requestInfo, resource) || !userCanUpdateResource(requestInfo, resource)
+                   ? produceRemovedPublicationResponse(resource, requestInfo)
+                   : producePublicationResponse(requestInfo, resource);
```

Comment:
1. We should redirect when user is asking for redirect.
2. We should return tombstone without files when user does not want to redirect or user has no righta to see files for unpublished resource.
3. We should serve unpublished resource with files when user has rights and does not want to redirect.